### PR TITLE
Lon 521

### PR DESCRIPTION
**Pull request by vmartos**
- Fixes Previous method.
- The playing state doesn't change when previous or next button are pressed.
- Add test to control the playing state on next and previous.

### CODE REVIEW

File: LongoMatch.Services/PlayerController.cs [hunk 1]
```diff
@@ -663,12 +663,17 @@ public void Previous (bool force = false)
 				if (loadedPlaylistElement is PlaylistPlayElement && !force) {
 					TimelineEvent play = (loadedPlaylistElement as PlaylistPlayElement).Play;
 					if ((CurrentTime - play.Start).MSeconds > 500) {
-						Seek (new Time (0), true);
+						AbsoluteSeek (play.Start, true);
+						return;
+					}
+				} else if ((loadedPlaylistElement is PlaylistVideo || loadedPlaylistElement is PlaylistImage) && !force) {
+					if ((CurrentTime - loadedSegment.Start).MSeconds > 500) {
```
Comment (vguzmanp) [inline LongoMatch.Services/PlayerController.cs]: We should check for this (and the `force`parameter) only once

**Comment by vguzmanp** [comment]
I have found a weird behaviour: If you have a presentation with an external video at the end, and the playback reaches the end of the video, the playhead goes to the start of the video, still in `Playing` state, and the video shows "Out of scope"

File: LongoMatch.Services/PlayerController.cs [hunk 2]
```diff
@@ -660,15 +660,17 @@ public void Previous (bool force = false)
 				Seek (new Time (0), true);
 			} else if (loadedPlaylistElement != null) {
 				/* Select the start of the element if we haven't played 500ms, unless forced */
-				if (loadedPlaylistElement is PlaylistPlayElement && !force) {
-					TimelineEvent play = (loadedPlaylistElement as PlaylistPlayElement).Play;
-					if ((CurrentTime - play.Start).MSeconds > 500) {
-						Seek (new Time (0), true);
+				if (!force && (CurrentTime - loadedSegment.Start).MSeconds > 500) {
```
Comment (vguzmanp) [inline LongoMatch.Services/PlayerController.cs]: loadedSegment.Start == play.Start?
If they aren't always the same, we need something like this:

    Time start;
    if (loadedPlaylistElement is PlaylistPlayElement)
        start = play.Start;
    else
        start = loadedSegment.Start;

    if (!force && (CurrentTime - start)

File: LongoMatch.Services/PlayerController.cs [hunk 3]
```diff
@@ -660,15 +660,17 @@ public void Previous (bool force = false)
...
+					if (loadedPlaylistElement is PlaylistPlayElement || loadedPlaylistElement is PlaylistVideo) {
+						AbsoluteSeek (loadedSegment.Start, true);
+						return;
+					} else if (loadedPlaylistElement is PlaylistImage) {
+						Config.EventsBroker.EmitPlaylistElementSelected (LoadedPlaylist, loadedPlaylistElement, Playing);
```
Comment (vguzmanp) [inline LongoMatch.Services/PlayerController.cs]: Couldn't we do a `EmitPlaylistElementSelected` in all cases?

File: LongoMatch.Services/PlayerController.cs [hunk 4]
```diff
@@ -909,7 +911,6 @@ void InternalOpen (MediaFileSet fileSet, bool seek, bool force = false, bool pla
 					} else {
 						player.Open (fileSet [0]);
 					}
-					EmitTimeChanged (new Time (0), player.StreamLength);
```
Comment (vguzmanp) [inline LongoMatch.Services/PlayerController.cs]: This updates the current time and the current duration. Do we update this elsewhere?

Comment (vmartos) [inline LongoMatch.Services/PlayerController.cs]: Yes, in Tick() method (runs in background) is always retrieving the current playback time to EmitTimeChanged in order to update the playhead in the Timerules

Comment (vguzmanp) [inline LongoMatch.Services/PlayerController.cs]: :+1:

**Pull request merged**